spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -315,16 +317,37 @@ private boolean obtainLock() {
 			 */
 			this.lockedAt = System.currentTimeMillis();
 			this.threadName = currentThread.getName();
-			Boolean success = RedisLockRegistry.this.redisTemplate.boundValueOps(
-					constructLockKey()).setIfAbsent(this);
+
+			Boolean success = RedisLockRegistry.this.redisTemplate.execute(new SessionCallback<Boolean>() {
+				@SuppressWarnings("unchecked")
+				@Override
+				public Boolean execute(RedisOperations ops) throws DataAccessException {
+					String key = constructLockKey();
+
+					ops.watch(key); //monitor key
+
+					ops.multi(); //transaction start
+
+					//can't relay on operations result inside transaction, execution is delayed till `exec()`
+					ops.opsForValue().setIfAbsent(key, RedisLock.this);
+
+					//set expire on key if exists
+					ops.expire(key, RedisLockRegistry.this.expireAfter, TimeUnit.MILLISECONDS);
+
+					//exec will contains all operations result or null - if execution was aborted due to 'watch'
```

Comment:
`exec contains`, not the future time. A fortiori: future time can't be with the single third person present tense rule.
And further: `or null if execution has been aborted due to 'watch'`. Present perfect to reconcile tense.
Isn't it ?